File: src/libs/ThirdParty.VBSuppliers/Properties/ExpediaRapid/SerializableClasses/Precancel/BookingItineraryResponseRoomRate.vb
Imports Newtonsoft.Json
Imports ThirdParty.VBSuppliers.ExpediaRapid.SerializableClasses.Search

Namespace ExpediaRapid.SerializableClasses.BookingItinerary
    Public Class BookingItineraryResponseRoomRate

        <JsonProperty("id")>
        Public Property RateID As String

        <JsonProperty("refundable")>
        Public Property IsRefundable As Boolean

        <JsonProperty("cancel_refund")>
        Public Property CancelRefund As CancelRefund

        <JsonProperty("merchant_of_record")>
        Public Property MerchantOfRecord As String

        <JsonProperty("amenities")>
        Public Property Amenities As New List(Of String)

        <JsonProperty("links")>
        Public Property RateLinks As New Dictionary(Of String, Link)

        <JsonProperty("cancel_penalties")>
        Public Property CancelPenalities As New List(Of CancelPenalty)

        <JsonProperty("deposit_policies")>
        Public Property DepositPolicies As New List(Of DepositPolicy)

        <JsonProperty("nightly")>
        Public Property NightlyRates As New List(Of List(Of Rate))

        <JsonProperty("stay")>
        Public Property StayRates As New List(Of Rate)

        <JsonProperty("fees")>
        Public Property OccupancyRateFees As New List(Of Fee)

        <JsonProperty("promotions")>
        Public Property Promotions As Promotions

    End Class
End Namespace

File: src/libs/iVectorOne.Suppliers/ExpediaRapid/SerializableClasses/Precancel/BookingItineraryResponseRoomRate.cs
namespace iVectorOne.CSSuppliers.ExpediaRapid.SerializableClasses.BookingItinerary
{
    using System.Collections.Generic;
    using Newtonsoft.Json;
    using iVectorOne.CSSuppliers.ExpediaRapid.SerializableClasses.Search;

    public class BookingItineraryResponseRoomRate
    {

        [JsonProperty("id")]
        public string RateID { get; set; }

        [JsonProperty("refundable")]
        public bool IsRefundable { get; set; }

        [JsonProperty("cancel_refund")]
        public CancelRefund CancelRefund { get; set; }

        [JsonProperty("merchant_of_record")]
        public string MerchantOfRecord { get; set; }

        [JsonProperty("amenities")]
        public List<string> Amenities { get; set; } = new List<string>();

        [JsonProperty("links")]
        public Dictionary<string, Link> RateLinks { get; set; } = new Dictionary<string, Link>();

        [JsonProperty("cancel_penalties")]
        public List<CancelPenalty> CancelPenalities { get; set; } = new List<CancelPenalty>();

        [JsonProperty("deposit_policies")]
        public List<DepositPolicy> DepositPolicies { get; set; } = new List<DepositPolicy>();

        [JsonProperty("nightly")]
        public List<List<Rate>> NightlyRates { get; set; } = new List<List<Rate>>();

        [JsonProperty("stay")]
        public List<Rate> StayRates { get; set; } = new List<Rate>();

        [JsonProperty("fees")]
        public List<Fee> OccupancyRateFees { get; set; } = new List<Fee>();

        [JsonProperty("promotions")]
        public Promotions Promotions { get; set; }

    }
}